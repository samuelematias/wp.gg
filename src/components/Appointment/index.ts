export {
    Appointment,
    AppointementProps,
} from './Appointment';
