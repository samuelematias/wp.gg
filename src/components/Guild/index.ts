export {
    Guild,
    GuildProps,
} from './Guild';
