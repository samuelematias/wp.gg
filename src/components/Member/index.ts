export {
    Member,
    MemberProps,
} from './Member';
