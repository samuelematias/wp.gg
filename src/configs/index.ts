export {
    COLLECTION_USERS,
    COLLECTION_APPOINTMENTS
} from './database';
