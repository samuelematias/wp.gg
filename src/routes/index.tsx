import {
    DefaultTheme,
    NavigationContainer
} from '@react-navigation/native';

import {
    SignIn
} from '../screens';

import { useAuth } from '../hooks';

import { AppRoutes } from './app.routes';

import { theme } from '../global/styles';

export function Routes() {
    const { user } = useAuth();

    const navTheme = {
        ...DefaultTheme,
        colors: {
            ...DefaultTheme.colors,
            background: theme.colors.secondary100,
        },
    };
    return (
        <NavigationContainer
            theme={navTheme}
        >
            {user?.id ? <AppRoutes /> : <SignIn />}
        </NavigationContainer>
    )
}
