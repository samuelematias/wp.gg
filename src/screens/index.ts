export * from './AppointmentCreate';
export * from './AppointmentDetails';
export * from './Guilds';
export * from './Home';
export * from './SignIn';
