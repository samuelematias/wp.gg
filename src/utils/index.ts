export { categories } from './categories';
export { IMAGE_PLACEHOLDER } from './constants';
