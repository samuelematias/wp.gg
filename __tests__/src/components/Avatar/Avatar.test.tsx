import 'react-native';
import React from 'react';

import {
    render,
} from '../../../../jest/helper';

import { Avatar } from '../../../../src/components';

const urlImage = 'https://facebook.github.io/react-native/docs/assets/favicon.png';

const renderComponent = <Avatar urlImage={urlImage} />;

describe('Avatar', () => {
    it('should render correctly and match the snapshot', () => {
        const { toJSON } = render(renderComponent);
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly, the image, when have valid url image', () => {
        const { getByTestId } = render(renderComponent);

        expect(getByTestId('Avatar.image.testId')).toBeTruthy();
    });

    it('should render correctly, the image placeholder, when have a invalid url image as empty', () => {
        const renderComponentWithInvalidUrlImageAsEmpty = <Avatar urlImage={''} />;
        const { getByTestId } = render(renderComponentWithInvalidUrlImageAsEmpty);

        expect(getByTestId('Avatar.image.placeholder.testId')).toBeTruthy();
    });

    it('should render correctly, the image placeholder, when have a invalid url image as null', () => {
        const renderComponentWithInvalidUrlImageAsNull = <Avatar urlImage={null} />;
        const { getByTestId } = render(renderComponentWithInvalidUrlImageAsNull);

        expect(getByTestId('Avatar.image.placeholder.testId')).toBeTruthy();
    });
})
