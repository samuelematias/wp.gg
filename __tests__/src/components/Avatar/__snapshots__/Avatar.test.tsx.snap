// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render correctly and match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#9146FF",
        "borderRadius": 8,
        "height": 49,
        "justifyContent": "center",
        "marginRight": 22,
        "width": 49,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://facebook.github.io/react-native/docs/assets/favicon.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 46,
          "width": 46,
        },
      ]
    }
    testID="avatar-image"
  />
</View>
`;
